jira/client.py

```diff
@@ -579,6 +581,27 @@ def __init__(
 
         self._fields_cache_value: Dict[str, str] = {}  # access via self._fields_cache
 
+    @contextlib.contextmanager
+    def disable_internal_user_lookup(self):
+        """Return a context manager which disables internal username searches.
+
+        This creates a context which turns off the internal username search
+        that various methods of the client use. This allows re-using a known
+        good username across multiple operations, such as one returned by
+        :py:meth:`JIRA.search_users`.
+
+        An example usage would be to search for a given user, then bulk assign
+        tickets to them using :py:meth:`JIRA.assign_issue`.
+
+        Return
+            contextlib._GeneratorContextManager
+        """
+        self._internal_user_lookup = False
```

Comment:
```suggestion
        :Example:

            .. code-block:: python
                # Bulk assigning tickets to a know user, saving internal lookup time
                my_user_id = "user_x"
                issue_list = ... # some list of issues
                with jira_client.disable_internal_user_lookup():
                    for issue in issue_list:
                        jira_client.assign_issue(issue, my_user_id )

        Return:
            contextlib._GeneratorContextManager
        """
        self._internal_user_lookup = False
```

You can checkout your function generated docs here:
https://jira--1301.org.readthedocs.build/api.html#jira.client.JIRA.disable_internal_user_lookup - you used the right syntax for cross-referencing a method.